### DISCUSSION TITLE
Exclusive Query Params

### DISCUSSION THREAD

**Discussion by gonzacastro** [fastapi/fastapi#9304] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
.-
```

### Description

Hey! I use FastAPI at work and an idea came up out of a requirement. It is about exclusive query params, I mean that it'd be useful for some specific cases if there'd be a way to specify a set of params that couldn't be sent all at once. So with this, you save the work to check that on your code. Example:
I have an endpoint '/students' and I can apply a filter either to look up by name or id, so I add the query params id and name but I don't want the request to be performed and inform it directly in the Swagger that you can't send those two params at once.
At least for me it'd be very useful, and I think that maybe it can help others too, so I raise this concern. I'm open to discuss! Thanks in advance.

### Operating System

Linux

### FastAPI Version

0.85

### Python Version

3.9.13

**Comment by YuriiMotov** [comment, marked as answer]
I think this is impossible. There is no such feature in openapi specification.

One way to achieve this is to create two separate endpoints.
To avoid code duplication, you can move business logic to separate function and use it in both endpoints.

Another way is to specify "exclusiveness" in description and implement this logic manually.
